src/serde.rs

```diff
@@ -0,0 +1,85 @@
+use serde::de::value::MapDeserializer;
+use serde::de::{Deserialize, Deserializer, Error, IntoDeserializer, MapAccess, Visitor};
+use serde::ser::{Serialize, SerializeMap, Serializer};
+
+use core::fmt::{self, Formatter};
+use core::hash::{BuildHasher, Hash};
+use core::marker::PhantomData;
+
+use crate::ListOrderedMultimap;
+
+impl<K, V, S> Serialize for ListOrderedMultimap<K, V, S>
+where
+    K: Serialize + Hash + Eq,
```

Comment:
Alphabetical order

Reply:
Ok, will order `impl`s as `Deserialize`, `IntoDeserialize`, `Serialize`, `Visitor`.

Reply:
I was referring to the trait bounds: Serialize + Hash + Eq.

The ordering of the traits and structs in the file doesn't need to be alphabetical, should be in the order that makes most sense.

Reply:
Ok, I restored the old order here but reordered the trait bounds alphabetically.